lightning/src/ln/channel.rs

```diff
@@ -6075,6 +6112,7 @@ impl<SP: Deref> Channel<SP> where
 pub(super) struct OutboundV1Channel<SP: Deref> where SP::Target: SignerProvider {
 	pub context: ChannelContext<SP>,
 	pub unfunded_context: UnfundedChannelContext,
+	pub unconfirmed_context: UnconfirmedChannelContext,
```

Comment:
I think keeping the `unfunded_context` around is a good idea.

Say we sent an outbound request to a peer, and the peer neither disconnects from us nor sends back an AcceptChannel message.
Then, we need a measure to deal appropriately with such stale OuboundChannel requests.

Reply:
I considered adding it to the `UnfundedChannelContext`. But the timer in that context is shared between Inbound and Outbound, while the introduced here is specific to Outbound only. If we introduce it to both, it might complicate things with peer disconnection handling. So, after giving it some thought, I decided against it.

Let me know what you think about this and if we should go about sharing the timer between both Channel contexts and handling mid-handshake disconnection for the Inbound Channel too.

Looking forward to your thoughts!

Reply:
@TheBlueMatt Any reason we shouldn't do this for inbound channels, too? IIUC, we'll still be in the unfunded state after sending the `accept_channel` message ourself.

Reply:
Also, regarding naming, if we decided on a separate struct specific to outbound channels, we should name it in that manner, i.e., `UnfundedOutboundChannelContext`, rather than to one specific (though only) use of it.

Reply:
Isn't the second timer reset upon reconnection while the original timer is not? We'd at least need a second timer, if not a struct.

Reply:
Not quite sure I follow what you mean by my reading of #2096. We currently wait `UNFUNDED_CHANNEL_AGE_LIMIT_TICKS` but drop the channel if the peer disconnects. The issue says that if the peer disconnects we should wait one tick before dropping the channel instead. Are you saying to keep the age-based approach as-is but drop the channel if the peer is disconnected on the _next_ timer tick (i.e., not waiting for one more?)

Reply:
Thanks a lot, @TheBlueMatt and @jkczyz, for your thorough discussion

@TheBlueMatt
A final confirmation before going to implementation. Are you suggesting completely dropping the added new timer, and keeping the OutboundV1Channel around for UNFUNDED_CHANNEL_AGE_LIMIT_TICKS even if the peer disconnects (which in the worst case would mean keeping a stale outbound channel request around for 1 hour)?